test/test_mps.py

```diff
@@ -18,23 +18,42 @@
 from collections import defaultdict
 from torch._six import inf
 from torch.nn import Parameter
+from torch.testing._internal import opinfo
 from torch.testing._internal.common_utils import \
-    (gradcheck, gradgradcheck, run_tests, TestCase, download_file,
-     TEST_WITH_UBSAN, dtype_abbrs)
+    (gradcheck, gradgradcheck, run_tests, TestCase, download_file, IS_CI,
+     TEST_WITH_UBSAN, dtype_abbrs, skipIfSlowGradcheckEnv, TEST_WITH_ASAN, suppress_warnings)
 from torch.testing import make_tensor
 from torch.testing._comparison import TensorLikePair
 from torch.testing._internal.common_dtype import get_all_dtypes, integral_types
 import torch.backends.mps
 from torch.distributions import Uniform, Exponential
 from functools import partial
 
-from torch.testing._internal.common_methods_invocations import op_db
-from torch.testing._internal.common_device_type import ops, instantiate_device_type_tests
+from torch.testing._internal.common_methods_invocations import (
+    op_db,
+    UnaryUfuncInfo,
+    ReductionOpInfo,
+    SpectralFuncInfo,
+    BinaryUfuncInfo,
+)
+from torch.testing._internal.common_device_type import ops, instantiate_device_type_tests, onlyMPS
 from torch.testing._internal.common_nn import NNTestCase
 import numpy as np
 import torch
 import torch.utils._pytree as pytree
 
+
+# Copied from `test_ops.py` for the purposes of duplicating `test_numpy_ref`
+_ref_test_ops = tuple(
```

Comment:
Does this really want to filter out elementwise unary, elementwise binary, reduction, and spectral operations? 

The reason those are excluded in test_ops.py is because they have more specific tests that run in their own test suites, like:

https://github.com/pytorch/pytorch/blob/d73d4aa7de953a3794593ac9e6d6b3a1ce514c3c/test/test_unary_ufuncs.py#L287

But if those test suites aren't run using the MPS device then it wouldn't be duplicative to include them in this ref test.

Reply:
I think the end goal should be that those tests also run on MPS, but that's definitely too large for this PR.

That's one reason I'm not thrilled having this live over in test_mps, tbh, see https://github.com/pytorch/pytorch/pull/87342#discussion_r1001112900

Either way, I'll tackle expanding the set of tests in a followup. All I wanted was to land a dang `where` test! :)

Reply:
Sounds good!